ql/src/java/org/apache/hadoop/hive/ql/plan/ConditionalResolverMergeFiles.java

```diff
@@ -512,7 +555,30 @@ private void setupWorkWhenUsingManifestFile(MapWork mapWork, List<FileStatus> fi
     mapWork.setUseInputPathsDirectly(true);
   }
 
-  private Map<FileStatus, List<FileStatus>> getManifestDirs(FileSystem inpFs, List<FileStatus> fileStatuses)
+  private void setupWorkWhenUsingCustomHandler(MapWork mapWork, Path dirPath,
+                                               StorageHandlerMergeProperties mergeProperties) throws ClassNotFoundException {
+    Map<String, Operator<? extends OperatorDesc>> aliasToWork = mapWork.getAliasToWork();
+    Map<Path, PartitionDesc> pathToPartitionInfo = mapWork.getPathToPartitionInfo();
+    Operator<? extends OperatorDesc> op = aliasToWork.get(dirPath.toString());
+    PartitionDesc partitionDesc = pathToPartitionInfo.get(dirPath);
+    Path tmpDir = mergeProperties.getTmpLocation();
```

Comment:
Can `mergeProperties` be null?

I saw earlier that sometimes we set it to null:
```
StorageHandlerMergeProperties mergeProperties = useCustomStorageHandler ?
            storageHandler.getStorageHandlerMergeProperties(ctx.getCustomStorageHandlerProps()) : null;
```

Reply:
`mergeProperties` is present when its used under storage handler which supports it. The boolean `customStorageHandler` if true should make sure that `mergeProperties`is not null.